processor/src/stack/mod.rs

```diff
@@ -66,6 +66,32 @@ impl Stack {
         Ok(self.trace[0][self.step])
     }
 
+    /// Get states from the stack, which includes both the
+    /// trace state and overflow table.
+    /// If n is not passed in, this returns all states.
+    pub fn get_states(&self, n: Option<usize>) -> Vec<Felt> {
+        let size = n.unwrap_or(usize::MAX);
+        let mut states: Vec<Felt> = Vec::new();
+        let trace_states = self.trace_state().to_vec();
+        let mut i = 0;
+        while i < cmp::min(trace_states.len(), size) {
+            states.push(trace_states[i]);
+            i += 1;
+        }
```

Comment:
This might print out more values than needed. `self.trace_state()` always returns an array of 16 values, even if current stack depth is less than that. So, we need an extra check to handle this case.